daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/Struct.scala

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf
+package data
+
+import com.daml.lf.data.Ref.Name
+
+final case class Struct[+X] private (sortedFields: ImmArray[(Ref.Name, X)]) extends NoCopy {
```

Comment:
This currently doesn’t appear to enforce uniqueness of fields. Given that we rely on this and have a lookup method, should it?

Reply:
I’ll leave it up to you, I don’t feel particularly strongly here.